### PR TITLE
Switch server to Javalin

### PR DESCRIPTION
**Pull request by floogulinc**
*Note:* This is a pull request onto #361 rather than master because it depends on things changed in #361.

### CODE REVIEW

**Comment by floogulinc** [comment]
Next thing to do here is add the todo functionality and maybe switch to using some mongodb

**Comment by wallerli** [comment]
All required lab 2 todo functions have been added, with test covering key features of the todos:
- The database test has examples of testing getting all todos and getting specific todo with id.
- The todo controller spec has examples of testing the two new features in todos: sort and limit.

This lab is basically about Angular in client. But do we want a more thorough test for the server?

File: server/src/main/resources/users.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
     "_id": "588935f51c55b55c75a84848",
     "name": "Roseann Roberson",
     "age": 23,
-    "company": "GINKLE",
-    "email": "roseannroberson@ginkle.com"
+    "company": "OHMNET",
+    "email": "roseannroberson@ohmnet.com"
```
Comment (floogulinc) [inline server/src/main/resources/users.json]: this change will effect the hash used for the avatar in #368 so we need to watch for this when we bring them together.

File: server/src/main/java/umm3601/user/Database.java [hunk 2]
```diff
@@ -1,37 +1,43 @@
...
- * Since we don't want to complicate this lab with a real database,
- * we're going to instead just read a bunch of user data from a
- * specified JSON file, and then provide various database-like
- * methods that allow the `UserController` to "query" the "database".
+ * Since we don't want to complicate this lab with a real database, we're going
+ * to instead just read a bunch of user data from a specified JSON file, and
+ * then provide various database-like methods that allow the `UserController` to
+ * "query" the "database".
  */
 public class Database {
```
Comment (floogulinc) [inline server/src/main/java/umm3601/user/Database.java]: Now that we have `TodoDatabase` we should refactor this to be `UserDatabase`

**Comment by floogulinc** [comment]
Other than maybe changing the `Database` class to `UserDatabase`, this looks good to me. Great work @wallerli.

File: server/src/test/java/umm3601/todo/TodoControllerSpec.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    ctx.clearCookieStore();
+
+    db = new TodoDatabase(Server.TODO_DATA_FILE);
+    TodoController = new TodoController(db);
+  }
+
+  @Test
+  public void GET_to_request_all_todos() throws IOException {
+    // Call the method on the mock controller
+    TodoController. getTodos(ctx);
```
Comment (kklamberty) [inline server/src/test/java/umm3601/todo/TodoControllerSpec.java]: Is this space after TodoController the convention that is usually used?

File: server/src/test/java/umm3601/todo/TodoControllerSpec.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+  private TodoController TodoController;
+  private static TodoDatabase db;
+
+  @BeforeEach
+  public void setUp() throws IOException {
+    ctx.clearCookieStore();
+
+    db = new TodoDatabase(Server.TODO_DATA_FILE);
+    TodoController = new TodoController(db);
```
Comment (kklamberty) [inline server/src/test/java/umm3601/todo/TodoControllerSpec.java]: Is it the convention to use a capitalized name for an instance of a controller? I would have guessed not, but would not be surprised if I'm wrong.

File: server/src/test/java/umm3601/todo/TodoControllerSpec.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  @Test
+  public void GET_to_request_all_todos() throws IOException {
```
Comment (kklamberty) [inline server/src/test/java/umm3601/todo/TodoControllerSpec.java]: Is this naming following a particular convention for naming? If not, I would suggest something more like test_GET_request_for_all_todos() or testGETrequestForAllTodos() - something that explains that this tests a get request that was asking for all the todos. I'm not sure about the 'to' in the name

**Comment by wallerli** [comment]
There is `NullPointerException` when I run these tests on VS code debugger. We might need to watch this after merged.

**Comment by kklamberty** [comment]
> There is `NullPointerException` when I run these tests on VS code debugger. We might need to watch this after merged.

**Comment by kklamberty** [comment]
> There is `NullPointerException` when I run these tests on VS code debugger. We might need to watch this after merged.

Was that a problem that existed before these recent changes that you know of or was that the first time you checked that? I have not run this code, so I was just looking for understandability, clarity, and consistency.

**Comment by floogulinc** [comment]
I am going to merge this, we can figure out the testing issue when we check the rest of the updates in the lab environment and fix bugs.

**Pull request merged**